### PR TITLE
[FL-3375] SubGhz: add CC1101 module external

### PR DESCRIPTION
**Pull request by Skorpionm**
# What's new

- [FL-3375] SubGhz: add CC1101 module external

# Verification 

- Connect the external module CC1101 according to the attached diagram and check the reception and transmission in the SubGhz application to the external and internal radio devices
<html>
<body>
<p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">To switch to an external radio module, go to<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--color-neutral-muted); border-radius: 6px;">menu -&gt; Sub-GHz -&gt; Radio Settings</code><span> </span>and switch<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--color-neutral-muted); border-radius: 6px;">Module</code><span> </span>to<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--color-neutral-muted); border-radius: 6px;">External</code></p>

Flipper Zero | СС1101
-- | --
2 | MOSI
3 | MISO
4 | CS
5 | SCK
6 | GD0
9 | VCC
11 | GND

</body>
</html>

https://user-images.githubusercontent.com/10090793/216795803-31a787c6-a19b-4368-8fcb-68438207683b.png

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

[FL-3375]: https://flipperzero.atlassian.net/browse/FL-3375?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: firmware/targets/f7/furi_hal/furi_hal_subghz.h [hunk 1]
```diff
@@ -5,6 +5,8 @@
 
 #pragma once
 
+#include <lib/subghz/devices/preset.h>
```
Comment (skotopes) [inline firmware/targets/f7/furi_hal/furi_hal_subghz.h:8]: Move to separate entity

Comment (skotopes) [reply firmware/targets/f7/furi_hal/furi_hal_subghz.h:8]: General idea is to simplify furi_hal_subghz that it doesn't know anything about specific presets.
Instead provide methods to load raw preset data.

On device level implement loading preset data by enum

File: lib/subghz/devices/cc1101_configs.h [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 #include <cc1101.h>
 
-static const uint8_t furi_hal_subghz_preset_ook_270khz_async_regs[][2] = {
+static const uint8_t subghz_device_cc1101_preset_ook_270khz_async_regs[][2] = {
```
Comment (skotopes) [inline lib/subghz/devices/cc1101_configs.h]: split h and c

File: lib/subghz/devices/devices.c [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+bool subghz_devices_is_connect(const SubGhzDevice* device) {
+    bool ret = false;
+    if(device && device->interconnect->is_connect) {
+        ret = device->interconnect->is_connect();
+    }
+    return ret;
+}
+
+void subghz_devices_reset(const SubGhzDevice* device) {
```
Comment (skotopes) [inline lib/subghz/devices/devices.c:54]: furi_assert(device)

**Review by skotopes — CHANGES_REQUESTED** [review]

File: lib/subghz/devices/devices.c (hunk 3 above)
Comment (Skorpionm) [reply lib/subghz/devices/devices.c:54]: + fix

File: lib/subghz/devices/cc1101_configs.h (hunk 2 above)
Comment (Skorpionm) [reply lib/subghz/devices/cc1101_configs.h]: + fix

File: firmware/targets/f7/furi_hal/furi_hal_subghz.h (hunk 1 above)
Comment (Skorpionm) [reply firmware/targets/f7/furi_hal/furi_hal_subghz.h:8]: +fix

**Review by skotopes — APPROVED** [review]

**Comment by saxonwhitee** [comment]
can i use a 355mhz antenna on the cc1101?

**Comment by skotopes** [comment]
@saxonwhitee who can stop you?

**Pull request merged**